dao/meta_data_dao.go

```diff
@@ -15,7 +15,7 @@ func (a *MetaDataDaoImpl) SubCollectionList(primaryCollection interface{}, limit
 	metadatas := make([]m.MetaData, 0, limit)
 	collection, err := m.NewRelationObject(primaryCollection, -1, DB.Debug())
 	if err != nil {
-		return nil, 0, err
+		return nil, 0, util.NewErrNotFound("application")
```

Comment:
I changed this to `return nil, 0, util.NewErrNotFound("metadata")` because of  function's return value ... but this is good question what is expected in this case (`"application type not found"` or `"metadata not found"`) when no metadatas are found for not existing application type (I think when you call `api/sources/v3.1/application_types/{{id}}/app_meta_data` there is no check if application type ID exists or not) ... final decision will affect all SubCollectionList methods

Reply:
I think we can return "metadata not found" and add some verification to the handler (before calling the SubCollectionList method) if the application type ID exists, and if not, then we can return "application type not found" ... but this can be solved in the next PR , what do you think? @lindgrenj6 @MikelAlejoBR @lpichler

Reply:
ok, so I had meeting with Libor and he explained me this and I hope I understand it better now, so yes @MikelAlejoBR, there should be an "application_type"